src/user/room.ts

```diff
@@ -583,9 +583,9 @@ export class Room extends Accessory implements Sayable {
   }
 
   public on (event: 'invite', listener: (this: Room, inviter: Contact, invitation: RoomInvitation) => void)               : this
-  public on (event: 'leave',  listener: (this: Room, leaverList:  Contact[], remover?: Contact) => void)                  : this
-  public on (event: 'join',   listener: (this: Room, inviteeList: Contact[], inviter:  Contact) => void)                  : this
-  public on (event: 'topic',  listener: (this: Room, topic:       string,    oldTopic: string, changer: Contact) => void) : this
+  public on (event: 'leave',  listener: (this: Room, leaverList:  Contact[], remover:  Contact, date: Date) => void)                  : this
```

Comment:
We have to keep the `remover?` as optional or we will introduce breaking changes to our library.

And so that you will have to make the `date?` optional too.